File: IT/Variabili, operatori, strutture di controllo/Esercizi/C++/esercizio27.cpp
/* 
file esercizio27.cpp
compilare con: g++ -std=c++11 esercizio27.cpp
*/

#include <iostream>

using std::cin;
using std::cout;
using std::endl;
using std::string;

int main() 
{
	cout << "Scrivi un numero: ";

	string s;
	getline(cin, s);

	if (s.find_first_not_of("0123456789-") != string::npos) {
		cout << "Numero non valido!" << endl;
		return 0;
	}

	int n = stoi(s);

	if (n > 0) {
		cout << n << " è positivo." << endl;
	}
	else
	{
		if (n == 0) {
			cout << "Hai digitato zero!" << endl;
		}
		else {
			cout << n << " è negativo." << endl;
		}
	}
}

File: IT/Variabili, operatori, strutture di controllo/Esercizi/Java/Esercizio27.java
/* file Esercizio27.java */

public class Esercizio27 {
	public static void main(String [] args) {
		String s = System.console().readLine("Scrivi un numero: ");

		if (!s.matches("[\\d]+")) {
			System.out.println("Numero non valido!");
			return;
		}

		int n = Integer.parseInt(s);

		if (n > 0) {
			System.out.println(n + " è positivo.");
		}
		else {
			if (n == 0) {
				System.out.println("Hai digitato zero!");
			}
			else {
				System.out.println(n + " è negativo.");
			}
		}
	}
}

